lib/app.js

```diff
@@ -3,30 +3,66 @@
 const app = require('@arkecosystem/core-container');
 
 class App {
-
-  start() {
+  start(options) {
     const api = app.resolvePlugin("api");
     const monitor = app.resolvePlugin("p2p");
 
-    api.http.events.on('response', request => {
-      this._log(request);
-    });
-    monitor.server._core.events.on('response', request => {
-      this._log(request);
-    });
+    options = this._applyDefaultOptions(options);
+
+    const logger = app.resolvePlugin("logger");
+    logger.debug(`DETECTIVE OPTIONS: ${options}`);
+
+    if (options.api) {
+      api.http.events.on('response', request => {
+        this._log(request, options);
+      });
+    }
+
+    if (options.p2p) {
+      monitor.server._core.events.on('response', request => {
+        this._log(request, options);
+      });
+    }
   }
 
-  _log(request) {
+  _applyDefaultOptions(options) {
```

Comment:
let's set them in defaults.js instead, where you assign them to `config` attribute in `index.js`, like so: https://github.com/ArkEcosystem/core/blob/master/packages/core-webhooks/lib/index.js#L10